src/main/java/uk/gov/hmcts/reform/sscs/callback/handlers/IssueFurtherEvidenceHandler.java

```diff
@@ -68,15 +68,35 @@ public void handle(CallbackType callbackType, Callback<SscsCaseData> callback) {
 
     private void issueFurtherEvidence(SscsCaseData caseData) {
         List<DocumentType> documentTypes = Arrays.asList(APPELLANT_EVIDENCE, REPRESENTATIVE_EVIDENCE, DWP_EVIDENCE, JOINT_PARTY_EVIDENCE, HMCTS_EVIDENCE);
-        List<FurtherEvidenceLetterType> allowedLetterTypes = Arrays.asList(APPELLANT_LETTER, REPRESENTATIVE_LETTER, JOINT_PARTY_LETTER);
-        documentTypes.forEach(documentType -> doIssuePerDocumentType(caseData, allowedLetterTypes, documentType));
+        List<FurtherEvidenceLetterType> allowedLetterTypes = Arrays.asList(APPELLANT_LETTER, REPRESENTATIVE_LETTER, JOINT_PARTY_LETTER, OTHER_PARTY_LETTER, OTHER_PARTY_REP_LETTER);
+        documentTypes.forEach(documentType -> doIssuePerDocumentType(caseData, allowedLetterTypes, documentType, null));
+
+        List<SscsDocument> groupedOtherPartyDocuments = groupUniqueOtherPartyDocumentsByOtherPartyId(caseData.getSscsDocument());
+
+        if (groupedOtherPartyDocuments != null && groupedOtherPartyDocuments.size() > 0) {
+            groupedOtherPartyDocuments.forEach(doc -> doIssuePerDocumentType(caseData, allowedLetterTypes, DocumentType.fromValue(doc.getValue().getDocumentType()), doc.getValue().getOriginalSenderOtherPartyId()));
+        }
+    }
+
+    private List<SscsDocument> groupUniqueOtherPartyDocumentsByOtherPartyId(List<SscsDocument> sscsDocuments) {
+
+        return sscsDocuments.stream()
+            .filter(doc -> OTHER_PARTY_EVIDENCE.getValue().equals(doc.getValue().getDocumentType()) || OTHER_PARTY_REPRESENTATIVE_EVIDENCE.getValue().equals(doc.getValue().getDocumentType()))
+            .filter(d -> "No".equals(d.getValue().getEvidenceIssued()))
+            .filter(distinctByKey(p -> p.getValue().getOriginalSenderOtherPartyId()))
```

Comment:
will this filter out documents with multiple of the same other party Id's?

Reply:
At this stage, we are just attempting to work out how many unique Other Parties require the '609_97' letter. These are only sent to other parties that have uploaded evidence that has not been issued yet.

Reply:
would a map to getOriginalSenderOtherPartyId then filter on non null values and then a distinct work better than what is written here?